include/seqan3/alphabet/composition.hpp

```diff
@@ -175,41 +175,41 @@ struct alphabet_composition :
     //!\brief An index sequence up to the number of contained letters.
     static constexpr auto positions = std::make_index_sequence<sizeof...(alphabet_types)>{};
 
-    //!\brief Implementation of to_integral().
+    //!\brief Implementation of to_rank().
     template <std::size_t ...idx>
-    constexpr integral_type to_integral_impl(std::index_sequence<idx...> const &) const
+    constexpr rank_type to_rank_impl(std::index_sequence<idx...> const &) const
     {
-        using seqan3::to_integral;
+        using seqan3::to_rank;
         if constexpr (sizeof...(idx) > 0)
         {
-            return to_integral(get<0>(*this)) +
-                   ((to_integral(get<idx + 1>(*this)) * cummulative_alph_sizes[idx]) + ...);
+            return to_rank(get<0>(*this)) +
+                   ((to_rank(get<idx + 1>(*this)) * cummulative_alph_sizes[idx]) + ...);
```

Comment:
This remark has nothing to do with the PR

----
Why the if branch?

```c++
return ( (to_rank(get<idx + 1>(*this)) * cummulative_alph_sizes[idx]) + ... + to_rank(get<0>(*this)));
```

You could further simplify that (if cummulative_alph_sizes[0] = 1)
```c++
return ( (to_rank(get<idx>(*this)) * cummulative_alph_sizes[idx]) + ... + 0);